source/main.cpp

```diff
@@ -365,160 +357,34 @@ WMMSettingsView::~WMMSettingsView()
 	delete fAuthorView;
 
 	delete Icon;
-
-	SaveSettings( &Settings);
 }
 
 //---------------------------------------------------
-//	Attached to window - set siblings target to this
+//	Attached to window - set siblings target
 //---------------------------------------------------
 void
 WMMSettingsView::AttachedToWindow()
 {
-	slidRadius->SetTarget( this);
-	slidCircles->SetTarget( this);
-	slidLineWidth->SetTarget( this);
-	slidLineSpace->SetTarget( this);
-	slidPulseRate->SetTarget( this);
-	Window()->SetPulseRate( Settings.pulse);
+	slidRadius->SetTarget(fDemoView);
+	slidCircles->SetTarget(fDemoView);
+	slidLineWidth->SetTarget(fDemoView);
+	slidLineSpace->SetTarget(fDemoView);
+	slidPulseRate->SetTarget(fDemoView);
 }
 
 //---------------------------------------------------
 //	Draw view
 //---------------------------------------------------
 void
-WMMSettingsView::Draw( BRect updateRect)
-{
-	demoRect = fDemoView->Bounds();
-
-	//demo background
-	fDemoView->SetHighColor(255, 255, 255, 255);
-	fDemoView->FillRoundRect(demoRect, 2, 2);
-	fDemoView->SetHighColor(0, 0, 0, 255);
-	fDemoView->StrokeRoundRect(demoRect, 2, 2);
-
-	InitDemo();
-}
-
-//---------------------------------------------------
-//	Draw demo
-//---------------------------------------------------
-void
-WMMSettingsView::DrawDemo()
+WMMSettingsView::Draw(BRect updateRect)
 {
-	float right = demoRect.right, bottom = demoRect.bottom;
-
-	if( demoCircle < Settings.circles)
-	{
-		float oldPen = fDemoView->PenSize();
-		fDemoView->SetPenSize(Settings.lwidth);
-		BRegion region = BRegion(BRect(1, 1, right - 1, bottom - 1));
-		fDemoView->ConstrainClippingRegion(&region);
-
-		fDemoView->SetHighColor(demoColor);
-		fDemoView->StrokeEllipse(BPoint(right / 2, bottom / 2), demoRadius, demoRadius);
-		demoRadius += ( Settings.lwidth / 2) + Settings.lspace;
-		demoColor.alpha += demoAlphaMod;
-		demoCircle++;
-
-		fDemoView->ConstrainClippingRegion(NULL);
-		fDemoView->SetPenSize(oldPen);
-	}
-	else
-	{
-		//demo background
-		fDemoView->SetHighColor(255, 255, 255, 255);
-		fDemoView->FillRoundRect(BRect(1, 1, right - 1, bottom - 1), 2, 2);
-		//SetHighColor( 0, 0, 0, 255);
-		//StrokeRoundRect( demoRect, 2, 2);
-
-		InitDemo();
-	}
+	fDemoView->Invalidate();
```

Comment:
I had to do this to fix this issue:
![broken rectangle](https://cloud.githubusercontent.com/assets/6434360/22085888/ebe727b4-dd8a-11e6-9959-6af473a5b954.png)
(after rubbing another window atop of the demo area)
Is there a better way to take care of the problem?